### PR TITLE
Read authentication information from .netrc

### PR DESCRIPTION
**Pull request by guw**
Workaround for #13709

### CODE REVIEW

File: tools/build_defs/repo/jvm.bzl [hunk 1]
```diff
@@ -59,6 +65,16 @@ _PASS_PROPS = (
 
 _FETCH_SOURCES_ENV_VAR = "BAZEL_JVM_FETCH_SOURCES"
 
+def _get_auth(ctx, urls): # duplicated from http.bzl
```
Comment (fmeum) [inline tools/build_defs/repo/jvm.bzl:68]: I think that moving this to `utils.bzl` would be useful, not only for this PR but also for other rules that want to reuse the same logic.

Comment (guw) [reply tools/build_defs/repo/jvm.bzl:68]: Agreed. Is it ok to do this as part of this PR?

Comment (fmeum) [reply tools/build_defs/repo/jvm.bzl:68]: I would personally say yes, but will defer to a team member. Cc @meteorcloudy

Comment (guw) [reply tools/build_defs/repo/jvm.bzl]: @fmeum noticed something:
```
    if srcurls and _should_fetch_sources_in_current_env(repository_ctx):
        repository_ctx.download(
            srcurls,
            srcpath,
            srcsha,
            canonical_id = repository_ctx.attr.canonical_id,
            auth = get_auth(repository_ctx, srcurls),
        )
```

Should this be:
```
canonical_id = repository_ctx.attr.canonical_id or get_default_canonical_id(repository_ctx, srcurls),
```

**Comment by meteorcloudy** [comment]
We are using `http_archive` directly from the bazel repo at 
https://github.com/bazelbuild/bazel/blob/bb5ff63d18abda883f2ccf7cf47e009ef6012420/repositories.bzl#L19-L19 instead of from `@bazel_tools`.

So the lockfile has to be updated.

File: tools/build_defs/repo/jvm.bzl (hunk 1 above)
Comment (fmeum) [reply tools/build_defs/repo/jvm.bzl]: Probably yes, I think I just missed this rule when I made the canonical ID change.

**Comment by guw** [comment]
> So the lockfile has to be updated.

Should I do this in this PR?
Are there instructions?

**Comment by meteorcloudy** [comment]
> Please run `bazel mod deps --lockfile_mode=update` to update your lockfile.

Just follow the error message. You probably also need to run 
https://github.com/bazelbuild/bazel/blob/dc706553ccc5fbab578d96348ed4ea5c35dfae97/src/MODULE.tools#L1-L3

**Comment by guw** [comment]
Updated the lock files. Can someone review if this is good?

**Comment by meteorcloudy** [comment]
The lockfile changes should be good. But can you check the test failure?

https://storage.googleapis.com/bazel-untrusted-buildkite-artifacts/018c3658-6f74-40ff-b908-b0b49b7c0687/src/test/shell/bazel/external_integration_test/shard_9_of_15/test_attempts/attempt_2.log

```
ERROR: Evaluation of query "deps(//zoo:ball-pit)" failed: preloading transitive closure failed: no such package '@@endangered//jar': No such attribute 'netrc'
```

**Review by fmeum — APPROVED** [review]

**Review by meteorcloudy — APPROVED** [review]
Thanks!

**Comment by brentleyjones** [comment]
@bazel-io flag

**Comment by iancha1992** [comment]
The changes in this PR have been included in [Bazel 7.1.0 RC1](https://releases.bazel.build/7.1.0/rc1/index.html). Please test out the release candidate and report any issues as soon as possible. If you're using Bazelisk, you can point to the latest RC by setting USE_BAZEL_VERSION=last_rc.
Thanks!

**Pull request closed without merging**